src/index.js

```diff
@@ -5,8 +5,18 @@ const TYPE_IMPORT = 'Import';
 
 const buildImport = template(`
   (new Promise((resolve) => {
-    require.ensure([], (require) => {
-      resolve(require(SOURCE));
+    require.ensure([SOURCE], (require) => {
+      var ns = require(SOURCE);
+      
+      if (ns && ns.__esModule) {
+        resolve(ns);
+      } else {
+        var wrap = Object.create(null);
```

Comment:
This needs to have a fallback for pre-ES5 environments that lack `Object.create(null)` (Promises are shimmable; `Object.create` is not)